packages/core/src/auth/sso/ssoAccessTokenProvider.ts

```diff
@@ -195,15 +205,28 @@ export abstract class SsoAccessTokenProvider {
      * Wraps the given function with telemetry related to the browser login.
      */
     protected withBrowserLoginTelemetry<T extends (...args: any[]) => any>(func: T): ReturnType<T> {
-        if (telemetry.spans.some(s => s.name === 'aws_loginWithBrowser')) {
-            // During certain flows, eg reauthentication, we are already running within a span (run())
-            // so we don't need to create a new one.
+        const run = (span: Metric<AwsLoginWithBrowser>) => {
+            span.record({
+                credentialStartUrl: this.profile.startUrl,
+                source: SsoAccessTokenProvider._authSource,
+            })
+
+            // Reset source in case there is a case where browser login was called but we forgot to set the source.
+            // We don't want to attribute the wrong source.
+            SsoAccessTokenProvider.authSource = 'unknown'
+
             return func()
         }
 
+        // During certain flows, eg reauthentication, we are already running within a span (run())
+        // so we don't need to create a new one.
+        const span = telemetry.spans.find(s => s.name === 'aws_loginWithBrowser')
+        if (span !== undefined) {
+            return run(span as unknown as Metric<AwsLoginWithBrowser>)
```

Comment:
not a blocker: does this not work (or whatever the right syntax is?

```suggestion
        if (span !== undefined && span instanceof Metric) {
            return run(span as Metric<AwsLoginWithBrowser>)
```

Reply:
could not get anything like this to work... maybe my brain is fried on type checking right now :(